app/javascript/components/student/editor/ExercismMonacoEditor.tsx

```diff
@@ -88,6 +87,27 @@ export function ExercismMonacoEditor({
         connection.onClose(() => disposable.dispose())
```

Comment:
But it's never closed? Turning this into an effect solves this.

Reply:
Ive now put it in an effect and the websocket properly.